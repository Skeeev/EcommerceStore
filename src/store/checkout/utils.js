import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
