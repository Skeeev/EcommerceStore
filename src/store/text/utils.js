export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
