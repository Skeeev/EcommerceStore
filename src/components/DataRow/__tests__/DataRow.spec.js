import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { DataRow } from '..';

describe(`${getComponentDisplayName(DataRow)} component`, () => {
  const renderComponent = props => shallow(<DataRow {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      label: 'dummy label',
      value: 'dummy value'
    });

    expect(component).toMatchSnapshot();
  });
});
