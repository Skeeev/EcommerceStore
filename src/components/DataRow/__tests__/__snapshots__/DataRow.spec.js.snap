// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataRow component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    dummy label
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    dummy value
  </Text>
</View>
`;
